docs/posts/towards-1.0-sdk.md

```diff
@@ -0,0 +1,146 @@
+# lakeFS SDK in 1.0
+
+<img src="file:./images/1.0-is-coming.jpg" height="500px"
+	alt="1.0 is coming: a double rainbow over a traffic jam in Tel Aviv"/>
+
+## Goals
+
+We are putting the finishing touches on the :sparkles: lakeFS 1.0 release
+:sparkles:.  Part of the reason 1.0 is so important is that it promises
+stability to lakeFS users and to lakeFS developers.
+
+lakeFS version numbers generally follow [Semantic
+Versioning](https://semver.org/).  Version **1.0.0** will be the first major
+version, so it will be the first version to promise _major version
+stability_.  A brief summary of semantic versioning: Starting with this
+version,
+
+* **Patch** version upgrades (for instance, from 1.0.0 to 1.0.1, or from
+  1.3.11 to 1.3.12) can fix bugs but cannot add new API capabilities.  All
+  programs that worked with the previous version continue to work with the
+  new version.  All programs that work with the new version and are
+  unaffected by the fixed bugs would work with the old version.
+* **Minor** version upgrades (for instance, from 1.0.0 to 1.1.0, or from
+  1.3.11 to 1.4.0) can add API capabilities but cannot change existing APIs.
+  All programs that worked with the previous version continue to work with
+  the new version.
+* **Major** version upgrades (for instance, from 1.4.0 to 2.0.0, or from
+  1.17.12 to 2.0.0) can add, remove, or change existing APIs entirely.  A
+  program might stop working.
+
+So it's a pretty important change!  Up until now, even a bump from 0.104.0
+to 0.105.0 could break old programs.  Indeed that happened.  Starting with
+1.0.0, a minor version that breaks old programs is a bug.
+
+## 1.0.0 is a **major** version upgrade!
+
+All guarantees here are for minor version upgrades.  The upgrade to 1.0.0 is
+of course a **major version upgrade**, and it will necessarily break some
+client programs.
+
+## Interface guarantees
```

Comment:
Do we like to specify that the S3 interface will change based on the compatibility with latest S3 interface?